integrations/cohere/src/haystack_integrations/components/embedders/cohere/document_embedder.py

```diff
@@ -124,7 +114,6 @@ def to_dict(self) -> Dict[str, Any]:
             input_type=self.input_type,
             api_base_url=self.api_base_url,
             truncate=self.truncate,
-            use_async_client=self.use_async_client,
```

Comment:
Can we ensure backwards compatibility for deserialization in the `from_dict` method and drop `use_async_client` if it is part of an older serialized file before calling `default_to_dict`?

Reply:
Good idea. I also added deserialization tests, which were missing.